### PR TITLE
[WIP] Generate codecov reports for PRs

### LINKED ISSUE #1557: Test coverage is not reported on each pr
[kind/bug]

## What versions of software are you using?
- Operating System: CI Setup
- Output of `odo version`: master

## How did you find it?
Check coverage report on each pr

## Actual behavior
No coverage report

## Expected behavior
Test coverage should be reported on each pr

### PR DESCRIPTION
**Pull request by kanchwala-yusuf**
1. Changes to .codecov.yml to enable code coverage reports
2. Changes to generate-coverage.sh for fixing errors

Fixes #1557

**What kind of PR is this?**
/kind bug

**What does does this PR do / why we need it**:
This PR enables code coverage reports in PRs

**Which issue(s) this PR fixes**:

Fixes #1557 

**How to test changes / Special notes to the reviewer**:
Codecoverage reports should be seen with PRs.

### CODE REVIEW

File: scripts/generate-coverage.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 set -e
 echo "" > coverage.txt
-go test -i -race ./cmd/odo
+#go test -i -race ./cmd/odo
```
Comment (girishramnani) [inline scripts/generate-coverage.sh]: could you please remove this commented code?

File: .codecov.yml [hunk 2]
```diff
@@ -1,10 +1,35 @@
+codecov:
+  require_ci_to_pass: no
+
 comment:
   behavior: default
```
Comment (girishramnani) [inline .codecov.yml]: default behavior updates the coverage info but when updating, the creation date of the comment will stay the same. There will be an edited link with a drop-down where you can check the times updated.
there is a `new` setting which deletes old and post new one. Would considering that be better @kanchwala-yusuf ?
cc @kadel

**Comment by kanchwala-yusuf** [comment]
@girishramnani, Forgot to add the WIP in the PR title. The changes in the PR are tentative and subject to change.

**Comment by kanchwala-yusuf** [comment]
This PR requires codecov to be enabled at the openshift org level. @amitkrout is following the due procedure for the same.

Once, that is done, this PR can progress ahead.

cc: @kadel , @amitkrout

**Comment by amitkrout** [comment]
> This PR requires codecov to be enabled at the openshift org level. @amitkrout is following the due procedure for the same. Once, that is done, this PR can progress ahead.

Created a jira task of priority Urgent in ```projects/DPP``` and shared it in ```fourum-dp-platform``` for a quick action

**Comment by amitkrout** [comment]
> Created a jira task of priority Urgent in projects/DPP and shared it in fourum-dp-platform for a quick action

@kanchwala-yusuf This is the last update i got on the jira 
```
Right after I posted this, I found an org-level setting where I had to explicitly grant CodeCov access to openshift/odo, so I enabled it. I think you should be good to go.
```
Seems the access has been granted. Can you please rebase the pr and check the expected output.

File: scripts/generate-coverage.sh [hunk 3]
```diff
@@ -17,4 +17,4 @@ for d in $(go list ./... | grep -v vendor | grep -v tests); do
         cat profile.out >> coverage.txt
         rm profile.out
     fi
-done
\ No newline at end of file
+done
```
Comment (amitkrout) [inline scripts/generate-coverage.sh]: Let's not add any space change to the original file.

File: .codecov.yml [hunk 4]
```diff
@@ -1,10 +1,35 @@
+codecov:
+  require_ci_to_pass: no
```
Comment (amitkrout) [inline .codecov.yml]: +1

File: scripts/generate-coverage.sh (hunk 3 above)
Comment (kanchwala-yusuf) [reply scripts/generate-coverage.sh]: done

**Comment by kanchwala-yusuf** [comment]
The code coverage is being generated as it has been enabled at the openshift org level. So, I guess we do not need this PR.

**Comment by kanchwala-yusuf** [comment]
So, closing this PR.

**Pull request closed without merging**